### ISSUE TITLE
Line breaks at generic types

### ISSUE THREAD

**Issue by mnazirov** [nicklockwood/SwiftFormat#1501]
Hello @nicklockwood ,
&nbsp; 
I have the following case where 0.48.11 SwiftFormat line breaks at generic types. Adding in the symbol exception didn't help
&nbsp; 
**Before formatting**
```
class SectionDatal<Model, Footer: UpdatableHeaderFooterView<Model>>: SectionDataProvider<Model, UpdatableHeaderFooterView<Model>, Footer> {
    ...
}
```
&nbsp; 

**After formatting**
```
class SectionData<Model, Footer: UpdatableHeaderFooterView<Model>>: SectionDataProvider<
    Model,
    UpdatableHeaderFooterView<Model>,
    Footer
> {
    ...
}
```
&nbsp; 

**Before formatting**
```
func createDestinationStep() -> DestinationStep<SomeFactory.ViewController, SomeFactory.Context> {
    ...
}
```
&nbsp; 

**After formatting**
```
func createDestinationStep() -> DestinationStep<
    SomeFactory.ViewController,
    SomeFactory.Context
> {
    ...
}
```
&nbsp; 

**Configurations**
```
# format options
--rules wrapArguments
--wraparguments before-first
--maxwidth 130
--nowrapoperators <
```

**Comment by nicklockwood** [comment]
@mnazirov that's a super-old version. Can you check that it's still happening on 0.51.15 before I investigate further? Thanks.

**Comment by mnazirov** [comment]
@nicklockwood Updated to version 0.51.15, but the issue with line breaks in generic types still persists
&nbsp;
Before formatting
```
func createConnectScreen() -> DestinationStep<NewConnectFactory.ViewController, NewConnectFactory.Context> {
   ...
}
```
&nbsp;
After formatting
```
func createConnectScreen() -> DestinationStep<
        NewConnectFactory.ViewController,
        NewConnectFactory.Context
> {
   ...
}
```

**Comment by nicklockwood** [comment]
@mnazirov thanks for double checking. The issue is that `--nowrapoperators` only applies to operators, and `<` isn't considered an operator by SwiftFormat when it's used as a bracket in a generics list.

It's not especially difficult for me to change that behavior, but I'm not sure if that would really solve your problem in this case, as it would just wrap at the `,` instead, which presumably isn't what you'd want.

What *is* the behavior you'd want in this case? Where should that line wrap?

**Comment by mnazirov** [comment]
@nicklockwood  Thank you for the detailed explanation. I've tried to ensure only the method parameters wrap, while the return types of generics remain unchanged.

**Comment by nicklockwood** [comment]
@mnazirov sure, I understand that. But in the example above there are no method parameters, so is your desired behavior that the line should just not wrap at all, even if it exceeds your specified max line width?

**Comment by mnazirov** [comment]
Yep, I want the line breaks to only occur at method arguments. Can you please advise how I can achieve this result?)

Before formatting
```
func createConnectScreen(argument: Int, argument2: Int) -> DestinationStep<NewConnectFactory.ViewController, NewConnectFactory.Context> {
   ...
}
```

Expected formatting result
```
func createConnectScreen(
    argument: Int, 
    argument2: Int
) -> DestinationStep<NewConnectFactory.ViewController, NewConnectFactory.Context> {
   ...
}
```

**Comment by nicklockwood** [comment]
@nicklockwood I think there's no way to do this currently other than to disable the `wrap` rule.
